app/assets/javascripts/snowcrash/modules/issues/table.js.coffee

```diff
@@ -1,218 +1,31 @@
-    else
-      console.log "The browser doesn't support local storage of settings."
-      console.log "Column selection can't be saved."
-
-  showHideColumns: =>
-    that = this
-
-    $.each @$table.find('thead th'), (index, th)->
-      $th = $(th)
-
-      if (column = $(th).data('column'))
-        that.tagColumnIndex ||= index if column == 'tags'
-        if that.selectedColumns.indexOf(column) > -1
-          that.$table.find("td:nth-child(#{index + 1})").css('display', 'table-cell')
-          $th.css('display', 'table-cell')
-        else
-          that.$table.find("td:nth-child(#{index + 1})").css('display', 'none')
-          $th.css('display', 'none')
-
-  storageKey: ->
-    @project ||= $('.brand').data('project')
-    "project.#{@project}.issue_columns"
-
 jQuery ->
   if $('body.issues.index').length
+    indexTable = new IndexTable('issue')
 
-    # Checkbox behavior: select all, show 'btn-group', etc.
-    $('.js-select-all-issues').click (e)->
-      $allCheckbox = $(this).find('input[type=checkbox]')
-      isChecked = $allCheckbox.prop('checked')
-      if e.target != $allCheckbox[0]
-        isChecked = !isChecked
-        $allCheckbox.prop('checked', isChecked)
-
-      $('input[type=checkbox].js-multicheck:visible').prop('checked', isChecked)
+    onMergeSelected = (event) =>
+      url = $(event.target).data('url')
+      issues_to_merge = []
 
-    # when selecting standalone issues, check if we must also check 'select all'
-    $('input[type=checkbox].js-multicheck').click ->
-      _select_all = $(this).prop('checked')
+      $('.js-items-table').find(indexTable.selectedItemsSelector).each ->
+        $row = $(this).parent().parent()
 
-      if _select_all
-        $('input[type=checkbox].js-multicheck').each ->
-          _select_all = $(this).prop('checked')
-          _select_all
+        id = @.name.split('_')[2]
+        issues_to_merge.push(id)
 
-      $('.js-select-all-issues > input[type=checkbox]').prop('checked', _select_all)
+      location.href = "#{url}?ids=#{issues_to_merge}"
 
     refreshToolbar = ->
       checked = $('input[type=checkbox]:checked.js-multicheck:visible').length
       if checked
-        $('.js-issue-actions').css('display', 'inline-block')
+        $('.js-table-actions').css('display', 'inline-block')
       else
-        $('.js-issue-actions').css('display', 'none')
+        $('.js-table-actions').css('display', 'none')
 
       if checked > 1
         $('#merge-selected').css('display', 'inline-block')
       else
         $('#merge-selected').css('display', 'none')
 
-    # on page load refresh toolbar
-    refreshToolbar()
-
-    # when selecting issues or 'select all', refresh toolbar buttons
-    $('.js-select-all-issues, input[type=checkbox].js-multicheck').click ->
-      refreshToolbar()
-
-
-
-    new IssueTable
+    # Handle the merge issues button click
+    $('#index-table').on('click', '#merge-selected', onMergeSelected)
```

Comment:
merge handlers and `onMergeSelected` could also be part of the  inherited class.

Reply:
✅